src/modifier/modifier.ts

```diff
@@ -2103,7 +2103,7 @@ export class PokemonInstantReviveModifier extends PokemonHeldItemModifier {
     });
 
     // Remove the Pokemon's FAINT status
```

Comment:
```suggestion
    // Remove any status the Pokemon had before fainting
```
(I tested and confirmed that without the status reset they don't get cleared, so it was needed for all status, not just faint)

Reply:
Whoops, I guess I forgot to update the comment.